fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/bulkexport/group/ChunkReader.java

```diff
@@ -0,0 +1,328 @@
+    String fhirSearchPatientGroupId;
+
+    /**
+     * Fhir search page size.
+     */
+    @Inject
+    @BatchProperty(name = "fhir.search.pagesize")
+    String fhirSearchPageSize;
+
+    @Inject
+    JobContext jobContext;
+
+    public ChunkReader() {
+        super();
+    }
+
+    private void fillChunkDataBuffer(List<Member> patientRefs) throws Exception {
+        TransientUserData chunkData = (TransientUserData) jobContext.getTransientUserData();
+        int compartmentPageNum = 1;
+        int resSubTotal = 0;
+        FHIRSearchContext searchContext;
+        Class<? extends Resource> resourceType = ModelSupport
+                .getResourceType(resourceTypes.get(indexOfCurrentResourceType));
+        if (chunkData != null) {
+            for (Member patientRef : patientRefs) {
+                if (patientRef == null) {
+                    continue;
+                }
+
+                String patientId =  patientRef.getEntity().getReference().getValue().substring(8);
+                Map<String, List<String>> queryParameters = new HashMap<>();
+
+                List<String> searchCreterial = new ArrayList<String>();
+
+                if (fhirSearchFromDate != null) {
+                    searchCreterial.add("ge" + fhirSearchFromDate);
+                }
+                if (fhirSearchToDate != null) {
+                    searchCreterial.add("lt" + fhirSearchToDate);
+                }
+                if (!searchCreterial.isEmpty()) {
+                    queryParameters.put(Constants.FHIR_SEARCH_LASTUPDATED, searchCreterial);
+                }
+
+                queryParameters.put("_sort", Arrays.asList(new String[] { Constants.FHIR_SEARCH_LASTUPDATED }));
+                searchContext = SearchUtil.parseQueryParameters("Patient", patientId,
+                        ModelSupport.getResourceType(resourceTypes.get(indexOfCurrentResourceType)), queryParameters, null, true);
+                do {
+                    searchContext.setPageSize(pageSize);
+                    searchContext.setPageNumber(compartmentPageNum);
+                    FHIRTransactionHelper txn = new FHIRTransactionHelper(fhirPersistence.getTransaction());
+                    txn.begin();
+                    FHIRPersistenceContext persistenceContext = FHIRPersistenceContextFactory.createPersistenceContext(null, searchContext);
+                    List<Resource> resources2 = fhirPersistence.search(persistenceContext, resourceType).getResource();
+                    txn.commit();
+                    compartmentPageNum++;
+
+                    for (Resource res2 : resources2) {
+                        if (res2 == null) {
+                            continue;
+                        }
+                        try {
+                            FHIRGenerator.generator(Format.JSON).generate(res2, chunkData.getBufferStream());
+                            chunkData.getBufferStream().write(Constants.NDJSON_LINESEPERATOR.getBytes());
```

Comment:
```suggestion
                            chunkData.getBufferStream().write(Constants.NDJSON_LINESEPARATOR.getBytes());
```
Why not make the NDJSON_LINESEPARATOR bytes anyway?

Reply:
avoiding getBytes each time?

Reply:
I didn't because I thought the cost is very minor, but make sense for me to make the change.